Wrappers/Python/cil/io/NEXUSDataWriter.py

```diff
@@ -100,6 +100,12 @@ def write(self):
         '''
         write dataset to disk
         '''   
+        # check filename and data have been set:
+        if self.file_name is None:
+            raise Exception('Path to nexus file to write to is required.')
+        if self.data is None:
+            raise Exception('Data to write out must be set.')
```

Comment:
I suppose these need to be `ValueError`

Reply:
The python docs define ValueError as:

> Raised when an operation or function receives an argument that has the right type but an inappropriate value, and the situation is not described by a more precise exception such as [IndexError](https://docs.python.org/3/library/exceptions.html#IndexError).

So I'm not sure that's the right one. Maybe TypeError (as it's None rather than being set to a path / data)

Reply:
Oh, thanks for pointing it out. It looks to me that we might want to use [`RuntimeError`](https://docs.python.org/3/library/exceptions.html#RuntimeError) then.

Reply:
I don't have stong feelings on the error. I think we just use  `Exception` in the other readers. Probably it should check it's not `None`, and raise a `TypeError`. Then check it can interpret it as a string, and raise a `TypeError`. And then check it can create the file at the right path and if not raise a `ValueError` or `RuntimeError`